### PR TITLE
ci: Add cni-dropgz compatibility to testing infrastructure

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

Adds in compatibility for "old" cni-dropgz images. 

To only be used for E2E Intergration pipelines.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: test/internal/kubernetes/utils_create.go [hunk 1]
```diff
@@ -344,7 +346,11 @@ func initCNSScenarioVars() (map[CNSScenario]map[corev1.OSName]cnsDetails, error)
 		log.Printf("%s not set to expected value \"ACN\", \"MCR\". Default to %s", envCNIImageRepo, imageRepoURL["ACN"])
 		url = imageRepoURL["ACN"]
 	}
-	initContainerNameCNI := path.Join(url, "azure-cni:") + os.Getenv(envCNIVersion)
+	cniImage := "azure-cni:"
+	if strings.Contains(strings.ToLower(envDROPGZ), "true") {
+		cniImage = "cni-dropgz"
```
Comment (vipul-21) [inline test/internal/kubernetes/utils_create.go:351]: should this be `cni-dropgz:` ?

**Comment by rbtr** [comment]
I'm really trying to kill this, why are you rezzing it ☹️

**Comment by jpayne3506** [comment]
Because a valid use case came up. We needed to validate 1.4.39.1 CNI. 👴

**Comment by rbtr** [comment]
> We needed to validate 1.4.39.1 CNI against the latest kernels. 👴 

For 1.4 validations should be updating the test infra on the release/1.4 branch, not bringing the 1.4 CNI to the 1.5 branch test infra.

**Comment by jpayne3506** [comment]
Closing as this PR should be directed to v1.4 and until we require latest kernel validation for previous releases it is not needed.

**Pull request closed without merging**